helix-core/src/main/java/org/apache/helix/task/AssignableInstanceManager.java

```diff
@@ -609,6 +611,19 @@ public void logQuotaProfileJSON(boolean onlyDisplayIfFull) {
     }
   }
 
+  /**
+   * For each JobType, record their total available threads across all instances
+   * @param clusterStatusMonitor
+   */
+  public void recordAvailableThreadsPerType(ClusterStatusMonitor clusterStatusMonitor) {
+    if (clusterStatusMonitor != null) {
+      for (String jobType : _globalThreadBasedQuotaMap.keySet()) {
+        JobMonitor jobMonitor = clusterStatusMonitor.getJobMonitor(jobType);
+        jobMonitor.updateAvailableThreadGauge((long) _globalThreadBasedQuotaMap.get(jobType));
```

Comment:
Question: Was this job type consistent with workflow type? I am not quite sure whether there could be some diverge of monitoring at job level against workflow level.

Reply:
They are the same thing: if workflow has a type, it applies to all jobs, otherwise job can have their own types. Since each job can have a different type, would `jobType` make more sense? If we refer to it as `workflowType` and use WorkflowMonitor, then each workflow can have different types of jobs and affects different WorkflowMonitors, which is confusing.

Reply:
I would suggest to use workflow type. We had the idea to hold different workflow type for different jobs, like a queue can handle some group of job types. Also make the workflow type could link it as the source of truth since the job type now is derived from workflow type. Logically, it could make more sense to let workflow level reporting.

Reply:
If I recall the API behavior correclty, if workflowType is defined, I believe all jobs inherit their parent workflow's type. Even if the user sets the job type (in addition to setting the workflow type), the parent workflow's type will override the job type.

If workflowType is not set, then users can set a different type for each individual jobType. meaning each job can have its own type.

Since it's possible for jobs to have different types when the parent workflow's type is not defined, I don't think we can use the workflow type here?

Reply:
@junkaixue I don't really understand your suggestion. Are you saying to not use `_globalThreadBasedQuotaMap`? Or are you saying to use `workflowMonitor` instead of `jobMonitor`?

Reply:
@NealSun96 No need for this. I think @narendly is right. There would be meaningless to have workflow level aggregation of metrics since assignment is happened at job level.